From 85ed51363fa782ffd18e1ec77750de494a2680f1 Mon Sep 17 00:00:00 2001
From: Jianzheng Zhou <B38613@freescale.com>
Date: Mon, 2 Jul 2012 14:57:15 +0800
Subject: [PATCH 017/310] ENGR00215487:wifi:add support for pcie intel wifi

Add BOARD_WLAN_VENDOR to control configs for different wifi.
Default BOARD_WLAN_VENDOR=ATHEROS.For intel wifi,it should be
INTEL.

Signed-off-by: Jianzheng Zhou <B38613@freescale.com>
---
 imx6/imx6.mk                           |  6 ++++
 imx6/sabresd/SabreSDBoardConfigComm.mk | 50 ++++++++++++++++++++++------------
 imx6/sabresd/init.rc                   |  1 +
 3 files changed, 39 insertions(+), 18 deletions(-)

diff --git a/imx6/imx6.mk b/imx6/imx6.mk
index b15458a..033aa1e 100644
--- a/imx6/imx6.mk
+++ b/imx6/imx6.mk
@@ -110,6 +110,12 @@ PRODUCT_PACKAGES += \
 	psatUtil				\
 	wmiconfig
 
+# Intel PCIE wifi firmware
+PRODUCT_PACKAGES += \
+	iwlwifi-6000-4.ucode			\
+	iwlwifi-5000-5.ucode			\
+	iwlagn.ko
+
 # gps related lib
 PRODUCT_PACKAGES += \
 	gps.default				\
diff --git a/imx6/sabresd/SabreSDBoardConfigComm.mk b/imx6/sabresd/SabreSDBoardConfigComm.mk
index e4504d7..45d693e 100755
--- a/imx6/sabresd/SabreSDBoardConfigComm.mk
+++ b/imx6/sabresd/SabreSDBoardConfigComm.mk
@@ -8,24 +8,38 @@ include device/fsl/imx6/BoardConfigCommon.mk
 TARGET_BOOTLOADER_BOARD_NAME := SABRELITE
 
 # Wifi
-BOARD_WLAN_DEVICE := ar6003
-BOARD_HAS_ATH_WLAN := true
-BOARD_WLAN_ATHEROS_SDK := system/wlan/atheros/compat-wireless
-BOARD_WPA_SUPPLICANT_DRIVER := NL80211
-BOARD_HOSTAPD_DRIVER := NL80211
-WPA_SUPPLICANT_VERSION := VER_0_9_ATHEROS
-HOSTAPD_VERSION := VER_0_9_ATHEROS
-WIFI_DRIVER_MODULE_PATH         := "/system/lib/modules/ath6kl_sdio.ko"
-WIFI_DRIVER_MODULE_NAME         := "ath6kl_sdio"
-WIFI_DRIVER_MODULE_ARG          := "suspend_mode=3 ath6kl_p2p=1"
-WIFI_DRIVER_P2P_MODULE_ARG      := "suspend_mode=3 ath6kl_p2p=1 debug_mask=0x2413"
-WIFI_SDIO_IF_DRIVER_MODULE_PATH := "/system/lib/modules/cfg80211.ko"
-WIFI_SDIO_IF_DRIVER_MODULE_NAME := "cfg80211"
-WIFI_SDIO_IF_DRIVER_MODULE_ARG  := ""
-WIFI_COMPAT_MODULE_PATH  := "/system/lib/modules/compat.ko"
-WIFI_COMPAT_MODULE_NAME  := "compat"
-WIFI_COMPAT_MODULE_ARG   := ""
-WIFI_TEST_INTERFACE      := "sta"
+BOARD_WLAN_VENDOR 			 := ATHEROS
+# for atheros vendor
+ifeq ($(BOARD_WLAN_VENDOR),ATHEROS)
+BOARD_WLAN_DEVICE			 := ar6003
+BOARD_HAS_ATH_WLAN 			 := true
+BOARD_WLAN_ATHEROS_SDK			 := system/wlan/atheros/compat-wireless
+WPA_SUPPLICANT_VERSION			 := VER_0_9_ATHEROS
+HOSTAPD_VERSION				 := VER_0_9_ATHEROS
+WIFI_DRIVER_MODULE_PATH          	 := "/system/lib/modules/ath6kl_sdio.ko"
+WIFI_DRIVER_MODULE_NAME          	 := "ath6kl_sdio"
+WIFI_DRIVER_MODULE_ARG           	 := "suspend_mode=3 ath6kl_p2p=1"
+WIFI_DRIVER_P2P_MODULE_ARG       	 := "suspend_mode=3 ath6kl_p2p=1 debug_mask=0x2413"
+WIFI_SDIO_IF_DRIVER_MODULE_PATH  	 := "/system/lib/modules/cfg80211.ko"
+WIFI_SDIO_IF_DRIVER_MODULE_NAME  	 := "cfg80211"
+WIFI_SDIO_IF_DRIVER_MODULE_ARG   	 := ""
+WIFI_COMPAT_MODULE_PATH			 := "/system/lib/modules/compat.ko"
+WIFI_COMPAT_MODULE_NAME			 := "compat"
+WIFI_COMPAT_MODULE_ARG			 := ""
+endif
+#for intel vendor
+ifeq ($(BOARD_WLAN_VENDOR),INTEL)
+BOARD_HOSTAPD_PRIVATE_LIB		 ?= private_lib_driver_cmd
+BOARD_WPA_SUPPLICANT_PRIVATE_LIB 	 ?= private_lib_driver_cmd
+WPA_SUPPLICANT_VERSION			 := VER_0_7_X_INTEL
+HOSTAPD_VERSION				 := VER_0_7_X_INTEL
+WIFI_DRIVER_MODULE_PATH          	 := "/system/lib/modules/iwlagn.ko"
+WIFI_DRIVER_MODULE_NAME          	 := "iwlagn"
+WIFI_DRIVER_MODULE_PATH			 ?= auto
+endif
+BOARD_WPA_SUPPLICANT_DRIVER      	 := NL80211
+BOARD_HOSTAPD_DRIVER             	 := NL80211
+WIFI_TEST_INTERFACE			 := "sta"
 
 BOARD_MODEM_VENDOR := AMAZON
 
diff --git a/imx6/sabresd/init.rc b/imx6/sabresd/init.rc
index 9e3ba7a..3a107c7 100755
--- a/imx6/sabresd/init.rc
+++ b/imx6/sabresd/init.rc
@@ -69,6 +69,7 @@ on boot
 
     # No bluetooth hardware present
     setprop hw.bluetooth 0
+    setprop wlan.interface wlan0
 
 service wpa_supplicant /system/bin/wpa_supplicant \
     -Dnl80211 -puse_p2p_group_interface=0  -e/data/misc/wifi/entropy.bin
-- 
1.8.0

